MbedTLS 3.6 fails building on Linux arm32

### Tested versions

- Reproducible in 4.3.dev (89850d55)

### System information

Fedora Linux 40 (KDE Plasma) - Wayland - Vulkan (Forward+) - dedicated AMD Radeon RX 7600M XT (RADV NAVI33) () - AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16 Threads)

### Issue description

While trying to build 4.3-dev6 binaries, I got a build error for Linux arm32:
```
gcc -o thirdparty/mbedtls/library/aesce.linuxbsd.editor.arm32.o -c -std=gnu11 -flto -pipe -O2 -w -isystem thirdparty/glad -DTOOLS_ENABLED -DDEBUG_ENABLED -DNDEBUG -DENGINE_UPDATE_CHECK_ENABLED -DNO_EDITOR_SPLASH -DSOWRAP_ENABLED -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DDBUS_ENABLED -DSPEECHD_ENABLED -DXKB_ENABLED -DJOYDEV_ENABLED -DUDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DX11_ENABLED -DLIBDECOR_ENABLED -DWAYLAND_ENABLED -DVULKAN_ENABLED -DRD_ENABLED -DGLES3_ENABLED -DCRASH_HANDLER_ENABLED -DMINIZIP_ENABLED -DBROTLI_ENABLED -DTHREADS_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DVK_USE_PLATFORM_WAYLAND_KHR -DGLAD_ENABLED -DEGL_ENABLED -DGODOT_MODULE -DMBEDTLS_CONFIG_FILE=\"thirdparty/mbedtls/include/godot_module_mbedtls_config.h\" -Ithirdparty/mbedtls/include -Ithirdparty/libpng -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zstd -Ithirdparty/zlib -Ithirdparty/clipper2/include -Ithirdparty/brotli/include -Iplatform/linuxbsd -Ithirdparty/linuxbsd_headers/wayland -Ithirdparty/linuxbsd_headers -Iplatform/linuxbsd -I. thirdparty/mbedtls/library/aesce.c
In file included from thirdparty/mbedtls/library/common.h:23,
                 from thirdparty/mbedtls/library/aesce.c:42:
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h: In function 'aesce_encrypt_block':
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:189:13: note: called from here
  189 |     block = vaeseq_u8(block, vld1q_u8(keys));
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17028:1: error: inlining failed in call to 'always_inline' 'vaesmcq_u8': target specific option mismatch
17028 | vaesmcq_u8 (uint8x16_t __data)
      | ^~~~~~~~~~
thirdparty/mbedtls/library/aesce.c:160:13: note: called from here
  160 |     block = vaesmcq_u8(block);                \
      |             ^~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:159:13: note: called from here
  159 |     block = vaeseq_u8(block, vld1q_u8(keys)); \
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/arm-godot-linux-gnueabihf_sdk-buildroot/lib/gcc/arm-godot-linux-gnueabihf/13.2.0/include/arm_neon.h:17014:1: error: inlining failed in call to 'always_inline' 'vaeseq_u8': target specific option mismatch
17014 | vaeseq_u8 (uint8x16_t __data, uint8x16_t __key)
      | ^~~~~~~~~
thirdparty/mbedtls/library/aesce.c:189:13: note: called from here
  189 |     block = vaeseq_u8(block, vld1q_u8(keys));
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -o thirdparty/mbedtls/library/asn1write.linuxbsd.editor.arm32.o -c -std=gnu11 -flto -pipe -O2 -w -isystem thirdparty/glad -DTOOLS_ENABLED -DDEBUG_ENABLED -DNDEBUG -DENGINE_UPDATE_CHECK_ENABLED -DNO_EDITOR_SPLASH -DSOWRAP_ENABLED -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DDBUS_ENABLED -DSPEECHD_ENABLED -DXKB_ENABLED -DJOYDEV_ENABLED -DUDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DX11_ENABLED -DLIBDECOR_ENABLED -DWAYLAND_ENABLED -DVULKAN_ENABLED -DRD_ENABLED -DGLES3_ENABLED -DCRASH_HANDLER_ENABLED -DMINIZIP_ENABLED -DBROTLI_ENABLED -DTHREADS_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DVK_USE_PLATFORM_WAYLAND_KHR -DGLAD_ENABLED -DEGL_ENABLED -DGODOT_MODULE -DMBEDTLS_CONFIG_FILE=\"thirdparty/mbedtls/include/godot_module_mbedtls_config.h\" -Ithirdparty/mbedtls/include -Ithirdparty/libpng -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zstd -Ithirdparty/zlib -Ithirdparty/clipper2/include -Ithirdparty/brotli/include -Iplatform/linuxbsd -Ithirdparty/linuxbsd_headers/wayland -Ithirdparty/linuxbsd_headers -Iplatform/linuxbsd -I. thirdparty/mbedtls/library/asn1write.c
scons: *** [thirdparty/mbedtls/library/aesce.linuxbsd.editor.arm32.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:02:03.59]
```

### Steps to reproduce

- Setup Linux arm32 build environment (e.g. using https://github.com/godotengine/build-containers or directly https://github.com/godotengine/buildroot/ binaries)
- `scons platform=linuxbsd arch=arm32 target=editor`

### Minimal reproduction project (MRP)

n/a

--- a/thirdparty/mbedtls/include/godot_module_mbedtls_config.h
+++ b/thirdparty/mbedtls/include/godot_module_mbedtls_config.h
@@ -49,8 +49,10 @@
 #undef MBEDTLS_DES_C
 #undef MBEDTLS_DHM_C
 
-#ifndef __linux__
+#if !(defined(__linux__) && defined(__aarch64__))
 // ARMv8 hardware AES operations. Detection only possible on linux.
+// May technically be supported on some ARM32 arches but doesn't seem
+// to be in our current Linux SDK's neon-fp-armv8.
 #undef MBEDTLS_AESCE_C
 #endif
 

